packages/button/src/Button/components/GenericButton.spec.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react"
+import { render, screen } from "@testing-library/react"
+import GenericButton, { CustomButtonProps } from "./GenericButton"
+
+it("renders a custom component when the 'component' prop is provided", () => {
+  render(<ExampleComponent />)
+  expect(
+    screen.getByRole("button", { name: "A custom button" })
+  ).toBeInTheDocument()
+})
+
+function ExampleComponent() {
+  const someRef = React.useRef()
+
+  return (
+    <GenericButton
+      label="A custom button"
+      ref={someRef}
+      href="https://cultureamp.design"
```

Comment:
This is a little confusing. Seeing `href` being declared makes me expect an anchor element, but the CustomComponent makes it a button. It's not really something anyone should be doing, so I don't think we should be making an example with.

Reply:
That's kinda the point here, the custom component's purpose is to override what you expect to be rendered. I think we probably disagree on something here: I don't think tests should be a place for best practice examples, that's what stories and tech docs are for.

Another factor is that the main purpose of this prop is for use with a router library component, which we're never going to pull in just for this. So we're never going to be able to make an example here that makes real world sense.

Reply:
I think if that's the case, then it would be better to go even more obscure and have the custom component be a whole different element? It took a bit of thinking to figure out why passing in a `href` resulted in a `button` (which is the main element the component renders) without looking at all the other props. As it is, if the `component` prop didn't work and the href check failed, wouldn't this still pass?